qcore/validate_vm.py

```diff
@@ -103,17 +106,42 @@ def vmfile(filename):
 #        # not as important, can be re-created based on params_vel.py
 #        pass
 
+    # 8: Check VM within bounds - only if DEM file is provided. If DEM file is not present, does not return false
+    if dem_path is None:
+        dem_path = DEM_PATH
+    if os.path.exists(dem_path):
+        with open(dem_path) as dem_fp:
+            dem_fp.next()
+            lat = dem_fp.next().split()
+            min_lat = float(lat[0])
+            max_lat = float(lat[-1])
+            lon = dem_fp.next().split()
+            min_lon = float(lon[0])
+            max_lon = float(lon[-1])
+        vel_crns_file = os.path.join(vm_dir, 'VeloModCorners.txt')
+        with open(vel_crns_file) as crns_fp:
+            next(crns_fp)
+            next(crns_fp)
+            for line in crns_fp:
+                lon, lat = map(float, line.split())
+                if lon < min_lon or lon > max_lon or lat < min_lat or lat > max_lat:
+                    return False, 'VM extents not contained within NZVM DEM'
+
     return True, 'VM seems alright: %s.' % (vm_dir)
 
 if __name__ == '__main__':
     rc = 1
+    parser = argparse.ArgumentParser()
+    parser.add_argument("VM_dir", type=str, help="path the VM folder")
+    parser.add_argument("-d", "--dem_path", type=str, help="path to the NZVM dem file, "
+                                                             "validates that the VM is within the bounds of the DEM")
+    args = parser.parse_args()
     try:
```

Comment:
Not sure. That's a @viktor76525 question. I'm guessing it's to do with return codes because this script is called as part of a workflow and if the VM fails validation the error is printed out and doesn't stop the execution of the parent script.